scripts/markdown_seo_lint.js

```diff
@@ -0,0 +1,219 @@
+const fs = require("fs");
+const yaml = require("js-yaml");
+const markdownlint = require("markdownlint");
+const path = require("path");
+
+const FRONT_MATTER_REGEX = /((^---\s*$[^]*?^---\s*$)|(^\+\+\+\s*$[^]*?^(\+\+\+|\.\.\.)\s*$))(\r\n|\r|\n|$)/m;
+
+function checkPageTitle(title) {
+    let error = null;
+    if (!title) {
+        error = "Missing page title";
+    }else if (title.trim().length === 0) {
+        error = "Page title is empty";
+    }else if (title.trim().length > 60) {
+        error = "Page title exceeds 60 characters";
+    }
+    return error;
+}
+
+function checkPageMetaDescription(meta) {
+    let error = null;
+    if (!meta) {
+        error = "Missing meta description";
+    }else if (meta.trim().length === 0) {
+        error = "Meta description is empty";
+    }else if (meta.trim().length < 50) {
+        error = "Meta description is too short. Must be at least 50 characters";
+    }else if (meta.trim().length > 160) {
+        error = "Meta descripiton is too long. Must be shorter than 160 characters";
+    }
+    return error;
+}
+
+// Recursively generate a list of markdown files from a given
+// list of paths.
+function searchForMarkdown(paths, result) {
+    // If the result arg does not exist we should create it.
+    if (!result) {
+        result = {
+            files: [],
+            frontMatter: {},
+        }
+    }
+    // Grab the first file in the list and generate
+    // its full path.
+    const file = paths[0];
+    const fullPath = path.resolve(__dirname, file);
+
+    // Check if the path is a directory
+    const isDirectory = fs.statSync(fullPath).isDirectory();
+
+    // Get the file suffix so we can grab the markdown files.
+    const fileParts = file.split(".");
+    const fileSuffix = fileParts[fileParts.length - 1];
+
+    // Ignore auto generated docs.
+    if (file.indexOf("/content/docs/reference/pkg") > -1) {
+        const remaining = paths.slice(1, paths.length);
+        return searchForMarkdown(remaining, result);
+
+    // If the path is a directory we want to add the contents of the directory
+    // to the list.
+    }else if (isDirectory) {
```

Comment:
Nit: Since the previous code block (`if (file.indexOf(....)`) returns, we can omit the `else` and make this a bit more readable.

```
// ...
// ...
if (isDirectory) {
     ...
}